onnxruntime/__init__.py

```diff
@@ -76,3 +76,71 @@
     __version__ = version
 
 onnxruntime_validation.check_distro_info()
+
+# Load nvidia libraries from site-packages/nvidia if the package is onnxruntime-gpu
+if (
+    __package__ == "onnxruntime-gpu"
+    # Just in case we rename the package name in the future
+    or __package__ == "onnxruntime-cuda"
+    or __package__ == "onnxruntime_gpu"
+    or __package__ == "onnxruntime_cuda"
+):
+    import ctypes
+    import os
+    import platform
+    import re
+    import site
+
+    # Get the site-packages path where nvidia packages are installed
+    site_packages_path = site.getsitepackages()[0]
+    nvidia_path = os.path.join(site_packages_path, "nvidia")
+    # Traverse the directory and subdirectories
+    if platform.system() == "Windows":  #
+        # Define the list of DLL patterns
+        cuda_libs = (
+            "cublas.dll",
+            "cudnn.dll",
+            "cudart.dll",
+            "nvrtc.dll",
```

Comment:
Please double check the filenames. For example, I did not see nvrtc.dll. In my machine, filenames are like:
cudart64_12.dll
nvrtc64_120_0.dll
cudnn64_9.dll

Reply:
Also use ldd to check the linux part.

Reply:
Constructed a regex pattern for each library name with optional parts

Reply:
> Please double check the filenames. For example, I did not see nvrtc.dll. In my machine, filenames are like: cudart64_12.dll nvrtc64_120_0.dll cudnn64_9.dll cublas64_12.dll
> 
> Maybe we only need preload those DLLs that show in dependency walker: [#22668 (comment)](https://github.com/microsoft/onnxruntime/issues/22668#issuecomment-2449106630)
> 
> curand.dll and nvJitLink.dll does not show up in the dependency walker. Please test whether we can exclude them from pre-load.

is cublasLT64_12.dll also needed?

Reply:
> is cublasLT64_12.dll also needed?

Yes.

Reply:
Added "cublasLt", to the `cuda_libs` (case is ignored)